python/cudf/cudf/tests/test_csv.py

```diff
@@ -1404,7 +1416,8 @@ def test_csv_writer_mixed_data(
         header=header,
         line_terminator=line_terminator,
         date_format="%Y-%m-%dT%H:%M:%SZ",
-        quoting=csv.QUOTE_NONNUMERIC,
+        quoting=csv.QUOTE_NONE,
+        escapechar="\\",
```

Comment:
Why do we need to change the Pandas behavior here?

Reply:
We are not adding quotes to strings when creating csv, and pandas adds quotes. to make it similar I had to introduce `QUOTE_NONE`, but with it needs `escapechar` to handle corner cases.

Reply:
For example,
```
(Pdb) df
   Integer       Date  Float  Integer2 String  Boolean
0     2345 1995-04-18  9.001      2345  Alpha     True
1    11987 1994-07-14  8.343       106   Beta    False
2     9027 2006-07-06  6.000      2088  Gamma     True
3     9027 2005-09-16  2.781    789277  Delta    False
(Pdb) columns
['Integer2', 'Float', 'Date', 'Integer', 'String', 'Boolean']
(Pdb) sep
';'
(Pdb) line_terminator
'NEWLINE'
(Pdb) df.to_csv(index=index, sep=sep, columns=columns, header=header, line_terminator=line_terminator, date_format="%Y-%m-%dT%H:%M:%SZ")

';"Integer2";Float;Date;"Integer";String;BooleanNEWLINE0;2345;9.001;1995-04-18T00:00:00Z;2345;Alpha;TrueNEWLINE1;106;8.343;1994-07-14T00:00:00Z;11987;Beta;FalseNEWLINE2;2088;6.0;2006-07-06T00:00:00Z;9027;Gamma;TrueNEWLINE3;789277;2.781;2005-09-16T00:00:00Z;9027;Delta;FalseNEWLINE'

```
You can observe that `Integer2` and `Integer` are being quoted but `Float` and `Date` are not. In case cudf, it doesn't quote any name..